homeassistant/components/switch/__init__.py

```diff
@@ -31,6 +32,7 @@
     discovery.services.BELKIN_WEMO: 'wemo',
     wink.DISCOVER_SWITCHES: 'wink',
     isy994.DISCOVER_SWITCHES: 'isy994',
+    camera.DISCOVER_SWITCHES: 'child_switch',
```

Comment:
Instead of teaching switch about the camera and that it has to link to child_switch, it would be better if we make a generic child_switch discovered entry so that any component could utilize child_switch without having to change the switch component.